pkg/nmi/server/server.go

```diff
@@ -245,7 +245,14 @@ func (s *Server) hostHandler(w http.ResponseWriter, r *http.Request) (ns string)
 	tokens, err := s.TokenClient.GetTokens(r.Context(), tokenRequest.ClientID, tokenRequest.Resource, *podID)
 	if err != nil {
 		klog.Errorf("failed to get service principal token for pod:%s/%s, error: %+v", podns, podname, err)
-		http.Error(w, err.Error(), http.StatusForbidden)
+		httpErrorCode := http.StatusForbidden
+		if !auth.IsTokenRefreshError(err) {
```

Comment:
From what I understand, if it's not a token refresh error, it's a health check error. Can we rename the function name to `IsHealthCheckError` to make it more explicit?

Reply:
With the current sdk thats a valid assumption because every sdk call is health check + token request. I can add `IsHealthCheckError` and internally call `IsTokenRefreshError` because token refresh error is the one that we can actually deduce.

Reply:
Sounds good